newIDE/app/src/EventsSheet/EventsTree/VariableDeclaration.js

```diff
@@ -0,0 +1,161 @@
+// @flow
+import { t, Trans } from '@lingui/macro';
+import { I18n } from '@lingui/react';
+import { type I18n as I18nType } from '@lingui/core';
+import * as React from 'react';
+import classNames from 'classnames';
+import { selectedArea, selectableArea, disabledText, icon } from './ClassNames';
+import { type ScreenType } from '../../UI/Responsive/ScreenTypeMeasurer';
+import { type WindowSizeType } from '../../UI/Responsive/ResponsiveWindowMeasurer';
+import {
+  shouldActivate,
+  shouldValidate,
+} from '../../UI/KeyboardShortcuts/InteractionKeys';
+import LocalIcon from '../../UI/CustomSvgIcons/ExternalEvents';
+import { getVariableTypeIcon } from '../ParameterFields/VariableField';
+
+const gd: libGDevelop = global.gd;
+
+const styles = {
+  container: {
+    whiteSpace: 'normal',
+    overflowWrap: 'anywhere', // Ensure everything is wrapped on small devices (or for long expressions).
+    cursor: 'pointer',
+    marginBottom: 1,
+  },
+};
+
+export const reactDndInstructionType = 'GD_DRAGGED_VARIABLE_DECLARATION';
+
+type Props = {|
+  variableName: string,
+  variable: gdVariable,
+  onClick: Function,
+  onDoubleClick: () => void,
+  selected: boolean,
+  disabled: boolean,
+
+  screenType: ScreenType,
+  windowSize: WindowSizeType,
+
+  id: string,
+|};
+
+const getVariableTypeLabel = (variable: gdVariable, i18n: I18nType) => {
+  const type = variable.getType();
+  return type === gd.Variable.String
+    ? i18n._('Text')
+    : type === gd.Variable.Number
+    ? i18n._('Number')
+    : type === gd.Variable.Boolean
+    ? i18n._('Boolean')
+    : type === gd.Variable.Structure
+    ? i18n._('Structure')
+    : type === gd.Variable.Array
+    ? i18n._('Array')
+    : 'unknown';
+};
+
+const getVariableValueAsString = (variable: gdVariable, i18n: I18nType) => {
+  const type = variable.getType();
+  return type === gd.Variable.Structure || type === gd.Variable.Array
+    ? i18n._(
+        variable.getChildrenCount() === 0
+          ? t`No children`
+          : variable.getChildrenCount() === 1
+          ? t`1 child`
+          : t`${variable.getChildrenCount()} children`
+      )
+    : type === gd.Variable.String
+    ? variable.getString()
+    : type === gd.Variable.Number
+    ? variable.getValue().toString()
+    : null;
+};
+
+export const VariableDeclaration = (props: Props) => {
+  const { variableName, variable, id } = props;
+  /**
+   * Render the different parts of the text of the instruction.
+   * Parameter can have formatting, be hovered and clicked. The rest
+   * has not particular styling.
+   */
+  const renderInstructionText = (i18n: I18nType) => {
```

Comment:
Nit: You can probably inline this entire function, it's not super useful here

Reply:
I don't know how to do this because the icon is a variable.